packages/salesforcedx-vscode-soql/test/vscode-integration/editor/soqlEditorInstance.test.ts

```diff
@@ -135,6 +135,42 @@ describe('SoqlEditorInstance should', () => {
     expect(updateDocumentSpy.getCall(0).args[1]).to.equal(aQuery);
   });
 
+  it('muffles the postMessage once if soql statement has NOT changed', async () => {
+    const postMessageSpy = sandbox.spy(mockWebviewPanel.webview, 'postMessage');
+    const aQuery = 'select a,b,c from somewhere';
+    instance.sendEvent({
+      type: MessageType.UI_SOQL_CHANGED,
+      payload: aQuery
+    });
+    // attempt to update webview with unchanged soql statement
+    instance.updateWebview(mockTextDocument);
+    expect(
+      postMessageSpy.callCount === 0,
+      `postMessageSpy callcount expected 0, but got ${postMessageSpy.callCount}`
+    );
+    // a second update with the same statement will send
+    instance.updateWebview(mockTextDocument);
```

Comment:
Not sure I completely understand this test. Shouldn't postMessageSpy call count still be 0 given there was no update to the query in the doc ?

Reply:
We only trap the message to updateWebview once.  So the call to sendEvent ( which mocks an incoming event and I just renamed mockReceiveEvent() to make more sense ) sets the internal cache of the soql statement to `select a,b,c from somewhere`.  

Then the first call to updateWebview is muffled, but not the second.  Even if the soql statement is unchanged, i just need to trap the first.  As a safety measure, i compare the statements and clear the cache after the first call.   

This potential gotcha with the change event is ( [warned by the docs](https://code.visualstudio.com/api/extension-guides/custom-editors#synchronizing-changes-with-the-textdocument) ).